src/coreclr/vm/jitinterface.cpp

```diff
@@ -7890,6 +7890,68 @@ CorInfoInline CEEInfo::canInline (CORINFO_METHOD_HANDLE hCaller,
         }
     }
 
+    // If the root level caller and callee modules do not have the same runtime
+    // wrapped exception behavior, and the callee has EH, we cannot inline.
+    _ASSERTE(!pCallee->IsDynamicMethod());
+    {
+        Module* pCalleeModule = pCallee->GetModule();
+        Module* pRootModule = pOrigCaller->GetModule();
+
+        if (pRootModule->IsRuntimeWrapExceptions() != pCalleeModule->IsRuntimeWrapExceptions())
+        {
+            if (pCallee->HasILHeader())
+            {
+                COR_ILMETHOD_DECODER header(pCallee->GetILHeader(), pCallee->GetMDImport(), NULL);
+                if (header.EHCount() > 0)
+                {
+                    result = INLINE_FAIL;
+                    szFailReason = "Inlinee and root method have different wrapped exception behavior";
+                    goto exit;
+                }
+            }
+            else if (pCallee->IsIL() && pCallee->GetRVA() == 0)
+            {
+                MethodInfoHelperContext cxt{ pCallee };
+                // We will either find or create transient method details.
+                _ASSERTE(!cxt.HasTransientMethodDetails());
+
+                // IL methods with no RVA indicate there is no implementation defined in metadata.
```

Comment:
This is copy&paste from `CEEInfo::getMethodInfo`. Can this be factored into a method to avoid duplicate code?